client.go

```diff
@@ -17,11 +18,12 @@ var _ error = StatusError{}
 type StatusError struct {
 	Status     string
 	StatusCode int
+	RetryAfter int
```

Comment:
any chance we could make this `time.Duration`, or add a comment that it is in seconds?